### PR TITLE
fixed input decimal fields - removed scientific notations that currently can be entered

### PR DESCRIPTION
**Pull request by aceazzemen**
for example e - + can be entered in decimal inputs. while technically correct, this is obviously not something that we want as an input from users.

### CODE REVIEW

File: src/components/input/Input.tsx [hunk 1]
```diff
@@ -222,14 +222,14 @@ export class Input extends React.Component<IInputProps, any> {
...
-    } else if (type === InputType.Email) {
+    if (type === InputType.Email) {
       return "email";
     } else if (type === InputType.PositiveIntegerText) {
       return "tel";
     } else {
-      return "number";
+      // "number" makes weird event.target.value, for example entering 123e will return an event with "" as the value,
+      // causing the filter we set above to do nothing. So its preferable to use text to let our filter work instead
+      return "text";
```
Comment (whipermr5) [inline src/components/input/Input.tsx]: Have you explored using `onInput`? https://github.com/facebook/react/issues/11142

Comment (whipermr5) [reply src/components/input/Input.tsx]: The problem is that our `onChange` handler is not firing; getting it to fire even while using `number` input should solve the problem

Comment (aceazzemen) [reply src/components/input/Input.tsx]: will try

File: src/components/input/__test__/Input.test.tsx [hunk 2]
```diff
@@ -118,7 +118,7 @@ describe("Input", () => {
         type={InputType.DecimalText}
       />
     );
-    expect(component.find("input").prop("type")).toEqual("number");
+    expect(component.find("input").prop("type")).toEqual("text");
```
Comment (whipermr5) [inline src/components/input/__test__/Input.test.tsx]: Update the test name too

Comment (aceazzemen) [reply src/components/input/__test__/Input.test.tsx]: done

**Pull request merged**